File: Ruby/season2/p007-009/p009.rb
# -*- coding: utf-8 -*-
=begin
　2017.4.27
　eachメソッド
=end

def main()
  # 配列宣言
  sarray = ["東京", "横浜", "大阪", "名古屋", "札幌", "京都", "メンソレータム", 30000000000000000000000000];
  p sarray;

  # ループ書き換え
  for i in 0..sarray.length
    sarray[i] = i;
  end
  p sarray

  # eachによる書き換え
  sarray.each do |i|
    sarray[i] = (1000-i).to_s;
  end
  p sarray

  # eachで取り出したもの中身
  i = 0
  sarray.each do |s|
    s = (i*10).to_s
    puts s
    puts sarray[i]
    i = i+1
  end
end


main()

File: Python/season2/p007-010/p009.py
# coding: UTF-8
"""
　2018.4.6
　三角形の二等辺三角形判定
"""

def isosceles(a, b, c):
  return (a==b) or (b==c) or (c==a)

a, b, c = 1, 2, 3
print( "a={} b={} c={} : {}".format(a, b, c, isosceles(a,b,c)) )

a, b, c = 1, 1, 3
print( "a={} b={} c={} : {}".format(a, b, c, isosceles(a,b,c)) )

a, b, c = 1, 2, 1
print( "a={} b={} c={} : {}".format(a, b, c, isosceles(a,b,c)) )

a, b, c = 15543.1, 15543.1, 3
print( "a={} b={} c={} : {}".format(a, b, c, isosceles(a,b,c)) )

a, b, c = 15543.12, 15543.12, 15543.12
print( "a={} b={} c={} : {}".format(a, b, c, isosceles(a,b,c)) )

a, b, c = 1234567.8901234567891, 1234567.8901234567892, 1234567.8901234567893
print( "a={} b={} c={} : {}".format(a, b, c, isosceles(a,b,c)) )